encoding/dabatch.go

```diff
@@ -0,0 +1,419 @@
+	batchBytes[0] = b.version
+	binary.BigEndian.PutUint64(batchBytes[1:], b.batchIndex)
+	binary.BigEndian.PutUint64(batchBytes[9:], b.l1MessagePopped)
+	binary.BigEndian.PutUint64(batchBytes[17:], b.totalL1MessagePopped)
+	copy(batchBytes[25:], b.dataHash[:])
+	copy(batchBytes[57:], b.blobVersionedHash[:])
+	copy(batchBytes[89:], b.parentBatchHash[:])
+	copy(batchBytes[121:], b.skippedL1MessageBitmap[:])
+	return batchBytes
+}
+
+// Hash computes the hash of the serialized DABatch.
+func (b *DABatchV1) Hash() common.Hash {
+	bytes := b.Encode()
+	return crypto.Keccak256Hash(bytes)
+}
+
+// BlobDataProof computes the abi-encoded blob verification data.
+func (b *DABatchV1) BlobDataProof() ([]byte, error) {
+	if b.blob == nil {
+		return nil, errors.New("called BlobDataProof with empty blob")
+	}
+	if b.z == nil {
+		return nil, errors.New("called BlobDataProof with empty z")
+	}
+
+	commitment, err := kzg4844.BlobToCommitment(b.blob)
+	if err != nil {
+		return nil, errors.New("failed to create blob commitment")
+	}
+
+	proof, y, err := kzg4844.ComputeProof(b.blob, *b.z)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create KZG proof at point, err: %w, z: %v", err, hex.EncodeToString(b.z[:]))
+	}
+
+	return BlobDataProofFromValues(*b.z, y, commitment, proof), nil
+}
+
+// Blob returns the blob of the batch.
+func (b *DABatchV1) Blob() *kzg4844.Blob {
+	return b.blob
+}
+
+// BlobVersionedHashes returns the blob versioned hashes of the batch.
+func (b *DABatchV1) BlobVersionedHashes() []common.Hash {
+	return []common.Hash{b.blobVersionedHash}
+}
+
+// BlobBytes returns the blob bytes of the batch.
+func (b *DABatchV1) BlobBytes() []byte {
+	return nil
+}
+
+// BlobDataProofForPointEvaluation computes the abi-encoded blob verification data.
+func (b *DABatchV1) BlobDataProofForPointEvaluation() ([]byte, error) {
+	if b.blob == nil {
+		return nil, errors.New("called BlobDataProofForPointEvaluation with empty blob")
+	}
+	if b.z == nil {
+		return nil, errors.New("called BlobDataProofForPointEvaluation with empty z")
+	}
+
+	commitment, err := kzg4844.BlobToCommitment(b.blob)
+	if err != nil {
+		return nil, errors.New("failed to create blob commitment")
+	}
+
+	proof, y, err := kzg4844.ComputeProof(b.blob, *b.z)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create KZG proof at point, err: %w, z: %v", err, hex.EncodeToString(b.z[:]))
+	}
+
+	return BlobDataProofFromValues(*b.z, y, commitment, proof), nil
+}
+
+// Version returns the version of the DABatch.
+func (b *DABatchV1) Version() uint8 {
+	return b.version
+}
+
+// SkippedL1MessageBitmap returns the skipped L1 message bitmap of the DABatch.
+func (b *DABatchV1) SkippedL1MessageBitmap() []byte {
+	return b.skippedL1MessageBitmap
+}
+
+// DataHash returns the data hash of the DABatch.
+func (b *DABatchV1) DataHash() common.Hash {
+	return b.dataHash
+}
+
+// DABatchV2 contains metadata about a batch of DAChunks.
+type DABatchV2 struct {
```

Comment:
For the other types (`DABlock` and `DAChunk`) it seems unnecessary but I think with `DABatch` we can see that there is a `DABatch` for many codecs. As such, I think this should live within each codec's "namespace" to increase cohesion and to make it easier to add a new codec later on (as it will make it clear which types will need to be implemented). 

For this I'd suggest moving each type into its codec's file or creating a file `codecv2_types.go` or similar to put the types. Then we can get rid of `dabatch.go`, `dablock.go` and `dachunk.go`.

Reply:
refactored in [7891503](https://github.com/scroll-tech/da-codec/pull/25/commits/7891503ea0e1b45e14a9426a0816873a510e10bb).